python/lsst/meas/base/forcedMeasurement.py

```diff
@@ -354,7 +354,14 @@ def run(self, measCat, exposure, refCat, refWcs, exposureId=None, beginOrder=Non
 
         # Create parent cat which slices both the refCat and measCat (sources)
         # first, get the reference and source records which have no parent
-        refParentCat, measParentCat = refCat.getChildren(0, measCat)
+        if 'deblend_nChild' in measCat.schema:
+            # Scarlet can create hierarcical blends, so we need a different
```

Comment:
hierarchical.. but anyway, is everything else here going to work as expected including both parents and I guess grandparents?

Reply:
Yes, because the children are set by the `getChildren` method, which returns the children for a given parent. That means that for depth=1 sources that also have children, the depth=2 children have no knowledge of their grandparent other than being able to trace it using `catalog.find(catalog.find(child["parent"])["parent"])`, for example.

Reply:
The NaN values that I was getting a few weeks ago were because I didn't include this extra check to add the depth=1 parents to the list of parents. So existing testing did verify that this works now.